lib/TopStoriesFeed.jsm

```diff
@@ -643,10 +655,24 @@ this.TopStoriesFeed = class TopStoriesFeed {
     return false;
   }
 
+  lazyLoadTopStories(dsPref) {
+    try {
+      this.discoveryStreamEnabled = JSON.parse(dsPref).enabled;
+    } catch (e) {
+      // Load activity stream top stories if fail to determine discovery stream state
+      this.discoveryStreamEnabled = false;
+    }
+    if (!this.discoveryStreamEnabled) {
+      this.initializeProperties();
+    }
+    this.init();
+  }
+
   async onAction(action) {
     switch (action.type) {
-      case at.INIT:
-        this.init();
+      // Check for pref initial values to lazy load activity stream top stories
+      case at.PREFS_INITIAL_VALUES:
```

Comment:
Oh this is neat. Probably worth a comment explicitly why we don't watch the usual INIT. As kinda implied in the other comment, there could be other actions that come in between INIT and PREFS_INITIAL_VALUES triggering something. I'm not sure if there are any other actions right now but nothing preventing some other feed from dispatching from their INIT either.

Reply:
I did confirm that there were no current actions between INIT and PREFS_INITIAL_VALUES:
```
"INIT"
"PREFS_INITIAL_VALUES"
"DISCOVERY_STREAM_CONFIG_SETUP"
"NEW_TAB_INIT"
"NEW_TAB_LOAD"
"NEW_TAB_STATE_REQUEST"
"NEW_TAB_INITIAL_STATE"
"UPDATE_SEARCH_SHORTCUTS"
"TELEMETRY_PERFORMANCE_EVENT"
"DISCOVERY_STREAM_LAYOUT_UPDATE"
"DISCOVERY_STREAM_SPOCS_ENDPOINT"
"DISCOVERY_STREAM_SPOCS_CAPS"
"DISCOVERY_STREAM_SPOCS_UPDATE"
"DISCOVERY_STREAM_SPOCS_FILL"
"DISCOVERY_STREAM_FEED_UPDATE"
```